iwyu.cc

```diff
@@ -3000,8 +2983,29 @@ class InstantiatedTemplateVisitor
   bool TraverseTemplateSpecializationTypeHelper(
       const clang::TemplateSpecializationType* type) {
     if (CanIgnoreCurrentASTNode())  return true;
-    if (CanForwardDeclareType(current_ast_node()))
-      current_ast_node()->set_in_forward_declare_context(true);
+
+    // Skip the template traversal if this occurrence of the template name is
+    // a class qualifier for an out of line method and not an actual type
+    // instantiation. Note also that in this case the method decl will not
+    // have any dependent types and so will be treated as non-templated code.
+    //
+    // Class<int>::method() {
+    // |-Type---^
+    // |-NNS------^
+    // |-CXXMethodDecl--^
+    //
+    ASTNode* const ast_node = current_ast_node();
```

Comment:
Can you spell this `const ASTNode* ast_node = ...` instead? That seems to be prevalent style.

Reply:
Not the same :) Anyway, I got rid of it. For such a small function it is noisy.